src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,226 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        this.tasks = new TaskList();
+        this.parser = new Parser();
+    }
+
+    public void run() {
+        ArrayList<String> lines =  this.storage.readFile();
+        for (String line : lines) {
+            try {
+                this.tasks.addTaskFromRepr(line);
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+            }
+        }
+        this.ui.init();
+        boolean b;
+        while ((b = this.parser.parseCommand(this.ui.readInput()))) {}
```

Comment:
Perhaps you can consider
1. naming boolean b with a variable name like `isRunning` to be more descriptive
2. write the code for reading input and parsing commands on multiple lines to aid readability

Let me know what you think!